### PR TITLE
DTSPO-15971 run tf plan on prod by default 

### PR DESCRIPTION
**Pull request by charlesluokb**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

DTSPO-15971
- run TF plan against prod by default on PRs
- users can opt out to specify 'not-plan-on-prod' as topic or label 

Notes:
*  could not write unit tests for 'withPipeline` as `gradlew test` is broken at moment
*
*

### CODE REVIEW

File: vars/withPipeline.groovy [hunk 1]
```diff
@@ -107,65 +106,42 @@ def call(type, String product, String component, Closure body) {
...
-              appPipelineConfig: pipelineConfig,
-              pipelineCallbacksRunner: callbacksRunner,
-              pipelineType: pipelineType,
-              subscription: subscription.nonProdName,
-              aksSubscription: aksSubscriptions.aat,
-              environment: environment.nonProdName,
-              product: product,
-              component: component,
-              tfPlanOnly: true
-            )
```
Comment (louisehuyton) [inline vars/withPipeline.groovy:121]: so its very essential these lines stay in, we need a tf plan of aat (since its the staging environment) to run every time as a separate step no matter what the topic/label is

File: vars/withPipeline.groovy [hunk 2]
```diff
@@ -107,65 +106,42 @@ def call(type, String product, String component, Closure body) {
...
-
+          onTerraformChangeInPR {
             def githubApi = new GithubAPI(this)
-
+            def prCache = githubApi.refreshPRCache() // e.g. demo, perftest, ithc, etc
```
Comment (louisehuyton) [inline vars/withPipeline.groovy:111]: so we need the for loop to go through these values because the user could name their branch anything it doesn't have to be an env name. For example: this_is_a_branch would try perform a tf plan against that value thinking its an environment

File: vars/withPipeline.groovy [hunk 3]
```diff
@@ -14,7 +14,6 @@ import uk.gov.hmcts.contino.PipelineCallbacksConfig
 import uk.gov.hmcts.contino.PipelineCallbacksRunner
 import uk.gov.hmcts.pipeline.AKSSubscriptions
 import uk.gov.hmcts.pipeline.TeamConfig
-import uk.gov.hmcts.pipeline.DeprecationConfig
```
Comment (louisehuyton) [inline vars/withPipeline.groovy:17]: this needs to stay - please add back in

File: build.gradle
Comment (louisehuyton) [inline build.gradle:1]: could you remove the entire file from the pr change please as this is unrelated

File: vars/withPipeline.groovy [hunk 4]
```diff
@@ -107,65 +106,42 @@ def call(type, String product, String component, Closure body) {
...
-              product: product,
-              component: component,
-              tfPlanOnly: true
-            )
-
+          onTerraformChangeInPR {
             def githubApi = new GithubAPI(this)
-
+            def prCache = githubApi.refreshPRCache() // e.g. demo, perftest, ithc, etc
+            def branchName = branch.branchName.toLowerCase() // could be PR-123 or #58
```
Comment (louisehuyton) [inline vars/withPipeline.groovy:112]: we use PR-123 instead of branchName because thats how it recognises the format of a PR. I don't think this is necessary.

File: vars/withPipeline.groovy (hunk 2 above)
Comment (louisehuyton) [reply vars/withPipeline.groovy:111]: think this may be fine - marking as resolved

File: build.gradle
Comment (charlesluokb) [reply build.gradle:1]: can not remove the entire file but I can revert the change I made in build.gradle. The file's already existed

File: vars/withPipeline.groovy (hunk 3 above)
Comment (charlesluokb) [reply vars/withPipeline.groovy:17]: The `DeprecationConfig` function is not referenced in the file hence I removed it. Can I know why it needs to stay

File: vars/withPipeline.groovy (hunk 1 above)
Comment (charlesluokb) [reply vars/withPipeline.groovy]: done

File: vars/withPipeline.groovy (hunk 3 above)
Comment (charlesluokb) [reply vars/withPipeline.groovy:17]: add it back in

**Review by endakelly — APPROVED** [review]
